packages/host/app/components/operator-mode-overlays.gts

```diff
@@ -0,0 +1,114 @@
+  embeddedCardElement: HTMLElement;
+}
+
+interface Signature {
+  Args: {
+    addToStack: (card: Card) => void;
+  };
+}
+
+export default class OperatorModeOverlays extends Component<Signature> {
+  <template>
+    {{#each this.overlayButtons as |overlayButton|}}
+      <button
+        {{on 'click' (fn this.args.addToStack overlayButton.model)}}
+        style='top: {{overlayButton.y}}px; left: {{overlayButton.x}}px; position: absolute; z-index: 2; border: none'
+        data-test-cardstack-operator-mode-overlay-button
+      >
+        Open
+      </button>
+    {{/each}}
+  </template>
+
+  @tracked overlayButtons = new TrackedArray<OverlayButton>([]);
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+
+    (globalThis as any)._CARDSTACK_OPERATOR_MODE_OVERLAYS_COMPONENT = this;
+
+    if (config.environment != 'test') {
+      this.refreshOverlayButtonCoordinates.perform();
+    }
+
+    registerDestructor(this, () => {
+      delete (globalThis as any)._CARDSTACK_OPERATOR_MODE_OVERLAYS_COMPONENT;
+      this.refreshOverlayButtonCoordinates.cancelAll();
+    });
+  }
+
+  registerElement(embeddedCardElement: HTMLElement, model: Card) {
+    schedule('afterRender', () => {
+      let { x, y } =
+        this.calculateOverlayedButtonCoordinates(embeddedCardElement);
+
+      this.overlayButtons.push({
+        x,
+        y,
+        model,
+        embeddedCardElement,
+      });
+    });
+  }
+
+  calculateOverlayedButtonCoordinates(parentElement: HTMLElement) {
+    let rect = parentElement.getBoundingClientRect();
+    return {
+      x: rect.x + rect.width - 60,
+      y: rect.y + 10,
+    };
+  }
+
+  refreshOverlayButtonCoordinates = task(async () => {
+    if (this.overlayButtons.length != 0) {
+      let newOverlayButtons = this.overlayButtons.map((overlayButton) => {
+        let { x, y } = this.calculateOverlayedButtonCoordinates(
+          overlayButton.embeddedCardElement
+        );
+
+        return {
+          x,
+          y,
+          model: overlayButton.model,
+          embeddedCardElement: overlayButton.embeddedCardElement,
+        };
+      });
+
+      if (
+        JSON.stringify(newOverlayButtons) != JSON.stringify(this.overlayButtons) // stringifying is an easy way to compare objects deeply
+      ) {
+        this.overlayButtons = new TrackedArray(newOverlayButtons);
+      }
+    }
+
+    // Refresh every 50ms. It seems like an appropriate amount of time to reposition the overlay buttons
+    // when resizing and interacting with cards in the stack. Probably could be more performant using
+    // some kind of an observer or a reactive mechanism -- but this is a quick and dirty solution for now which
+    // we can improve later.
+    await timeout(50);
+
+    this.refreshOverlayButtonCoordinates.perform();
+  });
+}
+
+declare module '@glint/environment-ember-loose/registry' {
+  export default interface Registry {
+    OperatorModeOverlays: typeof OperatorModeOverlays;
+  }
+}
```

Comment:
This approach (single global) is not going to work well when we have two stacks showing because it will not be clear which stack to open the card in.

Reply:
Good catch, that will be a problem. Perhaps we could figure it out from the card hierarchy and see which stack contains the parent card, but does sound a bit complicated. I'll look into the alternative approach you described

Reply:
The context now includes the stack. When we introduce multiple stacks we will know which one to use.